airflow/providers/amazon/aws/operators/emr.py

```diff
@@ -204,8 +224,12 @@ def __init__(
         self.wait_for_completion = wait_for_completion
         self.cluster_id = cluster_id
         self.aws_conn_id = aws_conn_id
-        self.waiter_countdown = waiter_countdown
-        self.waiter_check_interval_seconds = waiter_check_interval_seconds
+        self.waiter_max_attempts = (
+            (waiter_countdown // waiter_check_interval_seconds)
+            if waiter_max_attempts is NOTSET
+            else waiter_max_attempts
+        )
+        self.waiter_delay = waiter_check_interval_seconds if waiter_delay is NOTSET else NOTSET
```

Comment:
This logic could be way cleaner if NOTSET returned a falsy value.  Slack thread to discuss that change [here](https://apache-airflow.slack.com/archives/CCPRP7943/p1677618632397219).  Currently NOTSET does not have a `__bool__` method so `self.waiter_delay = waiter_delay or waiter_check_interval_seconds` would set `self.waiter_delay` to NOTSET.